elytron/src/main/java/org/wildfly/extension/elytron/SSLDefinitions.java

```diff
@@ -1511,6 +1532,56 @@ protected void installedForResource(ServiceController<SSLContext> serviceControl
         return createSSLContextDefinition(ElytronDescriptionConstants.CLIENT_SSL_CONTEXT, false, add, attributes, serverOrHostController);
     }
 
+    private static final AttributeDefinition[] DYNAMIC_SSL_CONTEXT_ATTRIBUTES = new AttributeDefinition[]{AUTHENTICATION_CONTEXT_ATTRIBUTE};
+
+    static ResourceDefinition getDynamicClientSSLContextDefinition(boolean serverOrHostController) {
+        return createSSLContextDefinition(ElytronDescriptionConstants.DYNAMIC_CLIENT_SSL_CONTEXT, false, new DynamicSSLContextAddHandler(), DYNAMIC_SSL_CONTEXT_ATTRIBUTES, serverOrHostController, Stability.COMMUNITY);
+    }
+
+    private static class DynamicSSLContextAddHandler extends BaseAddHandler {
+        private DynamicSSLContextAddHandler() {
+            super(SSL_CONTEXT_RUNTIME_CAPABILITY, DYNAMIC_SSL_CONTEXT_ATTRIBUTES);
```

Comment:
FYI, this BaseAddHandler super constructor (all of them, actually) relies on a deprecated (for removal) AbstractAddStepHandler constructor.

Reply:
@pferraro Good point, since we use it on many places and it is not related to this issue only, I have created jira https://issues.redhat.com/browse/WFCORE-6736 to address this

Reply:
Perfect, thanks.